homeassistant/components/darksky/weather.py

```diff
@@ -112,6 +114,12 @@ def setup_platform(
 class DarkSkyWeather(WeatherEntity):
     """Representation of a weather condition."""
 
+    _attr_native_precipitation_unit = LENGTH_MILLIMETERS
+    _attr_native_pressure_unit = PRESSURE_MBAR
```

Comment:
On line 186 it seems like this can be another unit?

```py
        if "us" in self._dark_sky.units:
            return round(convert_pressure(pressure, PRESSURE_HPA, PRESSURE_INHG), 2)
```

Reply:
Fixed in cb44a2a